### PR TITLE
21st Quattor workshop summary

### PR DESCRIPTION
**Pull request by jouvin**

### CODE REVIEW

File: _posts/2016-03-22-bruxelles-workshop.md [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+[Agenda](https://indico.cern.ch/event/491377/timetable/#20160322.detailed)
+
+
+## Site News
+
+Bruxelles
+
+* Moving the grid site to HTCondor
+* Very old SCDB desynchronized with mainstream SCDB: will restart a new one from scratch
+ * Or may be restart with Aquilon if it is ready for production with the template library
```
Comment (stdweird) [inline _posts/2016-03-22-bruxelles-workshop.md]: `may be` -> `maybe`

File: _posts/2016-03-22-bruxelles-workshop.md [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+
+
+## Configuration module status
+
+ncm-metaconfig: support for many services
+
+* No list available in the doc : look at the documentation
+* Plan to extract annotation from schema
+* moab: obsoletes ncm-maui and ncm-moab
+* One missing feature: ability to run a service-specific validation command on the configuration built before restarting the daemon
```
Comment (stdweird) [inline _posts/2016-03-22-bruxelles-workshop.md]: i created https://github.com/quattor/configuration-modules-core/issues/739

File: _posts/2016-03-22-bruxelles-workshop.md [hunk 3]
```diff
@@ -0,0 +1,280 @@
...
+
+* Soon: a next minor release with stuff currently merged in master
+* Later: a next major release with new metadata for escaping and the .tpl removal
+
+
+## Configuration module status
+
+ncm-metaconfig: support for many services
+
+* No list available in the doc : look at the documentation
```
Comment (stdweird) [inline _posts/2016-03-22-bruxelles-workshop.md]: If there's no list in the doc, what do you mean with "look at the documentation"?

File: _posts/2016-03-22-bruxelles-workshop.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+* New version testing: create a new snapshot, associate a few test nodes with the snapshot and deploy on them only
+
+clusters: set of different machines sharing the same configuration
+
+features: Aquilon looks for a template `config.pan` in the template directory (can contains several directories)
+
+* features are located in the  `features/` directory
+* Order of feature inclusions is not guaranteed but 2 stages (pre and post) in the inclusion process
+  * Create a meta-feature that includes other features in the appropriate order if necessary and the 2 stages are not enough
+  * Including featres in others features means that their usage cannot be tracked by Aquilon: normally they are added under /metadata in the profile
```
Comment (stdweird) [inline _posts/2016-03-22-bruxelles-workshop.md]: `features`

File: _posts/2016-03-22-bruxelles-workshop.md [hunk 5]
```diff
@@ -0,0 +1,280 @@
...
+* Interaction with Aquilon behind the scene
+
+
+## CAF
+
+Proposal to add support for event tracking (CAF::History, See [PR](https://github.com/quattor/CAF/pull/117))
+
+* Objective: makes components using CAF modules to be able to report about what was done by CAF
+  * For example track/report changed files
+  * Will also allow to remove files created by Quattor when a component is removed if desirable
```
Comment (stdweird) [inline _posts/2016-03-22-bruxelles-workshop.md]: this is not only about the Unconfigure, but also if e.g. a single service is removed from metaconfig (while other are still present).

File: _posts/2016-03-22-bruxelles-workshop.md [hunk 6]
```diff
@@ -0,0 +1,280 @@
...
+* CAF::History allows to build a list of event that can be processed later
+  * ncm-ncd [PR](https://github.com/quattor/ncm-ncd/pull/51) demonstrates the use of CAF::History to report all the touched files at the end of a component run
+  * Could imagine to flush the event list to a database at the end of a ncm-ncd run to keep track of modifications made by Quattor
+  * Nathan would prefer immediate reporting of the events rather than a deferred one (in case a component crashes before reporting occured): improve current implementation to track if an event has already been reported and makes possible to call the report method several times still reporting event once. A global property could be taken into account to decide wehn an event is added to the list whether immediate reporting should occur.
```
Comment (stdweird) [inline _posts/2016-03-22-bruxelles-workshop.md]: very long line.
typo:  `wehn` -> `when`

File: _posts/2016-03-22-bruxelles-workshop.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+
+* Pb: ncm-query format is undefined so difficult to assess compatibility
+* Discussion about `--format ncmquery`: agreement to remove it as it creates a dependency on this undefined format in the future
+  * In script, it is recommended to move to the json output of `ccm`
+  
+ncm-spma:
+
+* PR open to move YUM repo definition to a non standard directory (/etc/yum.quattor.repo.d), defined as the default one in /etc/yum.conf, to avoid problems due to RPMs installing repo file that will interfere with Quattor deployment later
+*  yumng plugin: agreement to merge it as soon as MS has merged the last changes
+* yum plugin: run `yum clean' for a repo about to be removed (else it will not be clean by later runs of `yum clean`)
```
Comment (stdweird) [inline _posts/2016-03-22-bruxelles-workshop.md]: missing backtick after `clean`

**Comment by stdweird** [comment]
@jouvin as usual, thanks a lot for this summary

**Comment by jrha** [comment]
21**st**

**Pull request merged**